src/sentry/rules/processing/delayed_processing.py

```diff
@@ -142,7 +142,9 @@ def get_condition_group_results(
             return None
 
         _, duration = condition_inst.intervals[unique_condition.interval]
-        comparison_interval = condition_inst.intervals[unique_condition.interval][1]
+        comparison_interval = condition_inst.intervals[
+            condition_data.get("comparisonInterval", "5m")
```

Comment:
So just keep in mind here that `condition_data` comes from the first condition we see for that `unique_condition`. So since we don't currently unique on different `comparisonInterval`, this will still produce incorrect results if we have two rules with conditions like

`% change of events in the last hour compared to 1 day ago`
`% change of events in the last hour compared to week day ago`

Whichever of these conditions is processed first will have  the comparison interval set, and the second will incorrectly compare to the previous conditions `comparisonInterval`.

I assume you're going to fix this bug separately so I'm ok with merging, but just wanted to make sure you're aware.

Reply:
oh yes this was just the easier of the 2 bugs to tackle first